code/io/joy-sdl.cpp

```diff
@@ -9,15 +9,66 @@
 using namespace io::joystick;
 using namespace os::events;
 
-namespace
-{
+namespace {
 typedef std::unique_ptr<Joystick> JoystickPtr;
 
 SCP_vector<JoystickPtr> joysticks;
 Joystick *currentJoystick = nullptr;
 
 bool initialized = false;
 
+/**
+ * @brief Compatibility conversion from HatPosition to array index
+ */
+inline
+int hatEnumToIdx(HatPosition in) {
+	return int(in - JOY_NUM_BUTTONS);
```

Comment:
Minor nitpick: `static_cast` or a C-style cast should be used here. Function style casts are potentially dangerous. This one isn't but it's better to keep the code consistent.